### ISSUE TITLE
The <sys/sysctl.h> header is deprecated and will be removed.

### ISSUE THREAD

**Issue by wojciechGaudnik** [sebastiencs/coreutils-icons#12]
In file included from lib/nproc.c:50:
/usr/include/x86_64-linux-gnu/sys/sysctl.h:21:2: error: #warning "The <sys/sysctl.h> header is deprecated and will be removed." [-Werror=cpp]
   21 | #warning "The <sys/sysctl.h> header is deprecated and will be removed."
      |  ^~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:10024: lib/nproc.o] Error 1
make[2]: Leaving directory '/home/bq666/Downloads/ls-icons'
make[1]: *** [Makefile:12675: all-recursive] Error 1
make[1]: Leaving directory '/home/bq666/Downloads/ls-icons'
make: *** [Makefile:6835: all] Error 2

**Comment by irfanhilmi000** [comment]
same

**Comment by irfanhilmi000** [comment]
same

**Comment by coaxial** [comment]
I have this error too when compiling.

**Comment by omentic** [comment]
My fork compiles, although I had to reset it onto coreutils upstream after trying and failing to merge. [It's available here.](https://github.com/j-james/ls-icons)

FWIW, I was also able to compile the current repository. Did you all `export CC=clang CXX=clang++` before `./configure`ing?

**Comment by coaxial** [comment]
> My fork compiles, although I had to reset it onto coreutils upstream after trying and failing to merge. [It's available here.](https://github.com/j-james/ls-icons)
> 
> FWIW, I was also able to compile the current repository. Did you all `export CC=clang CXX=clang++` before `./configure`ing?

No, if settings these variables explicitly, then it fails configuring because it can't find the compiler (I don't remember the exact error message off the top of my head, but it would only configure without exporting these two vars)
